services/src/main/java/io/grpc/protobuf/service/ProtoReflectionService.java

```diff
@@ -64,31 +65,38 @@
 import java.util.Queue;
 import java.util.Set;
 
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.GuardedBy;
+
 /**
  * Provides a reflection service for Protobuf services (including the reflection service itself).
  *
- * <p>Throws an exception if the set of services contain multiple protobuf files with declarations
- * of the same service, method, type, or extension.
+ * <p>Separately tracks mutable and immutable services. Throws an exception if either group of
+ * services contains multiple Protobuf files with declarations of the same service, method, type, or
+ * extension.
  */
 @ExperimentalApi("https://github.com/grpc/grpc-java/issues/2222")
 public final class ProtoReflectionService extends ServerReflectionGrpc.ServerReflectionImplBase
```

Comment:
Why? How would the service be added to a server then? The current use ([here](https://github.com/grpc/grpc-java/blob/b5b9ba560486faf25f874146e23159a8ed7437d2/services/src/test/java/io/grpc/protobuf/service/ProtoReflectionServiceTest.java#L92)) is to instantiate a new instance and add it the server.